gcc/testsuite/rust/compile/issue-2324-2.rs

```diff
@@ -0,0 +1,19 @@
+enum State {
+    Succeeded,
+    Failed { x: u32 },
+}
+
+fn print_on_failure(state: &State) {
+    match *state {
+        State::Succeeded => (),
+        State::Failed => (), // { dg-error "expected unit struct, unit variant or constant, found tuple variant" }
```

Comment:
should be `found struct variant`

Reply:
I believe this should say "found unit variant"
Since State::Failed is a unit variant, but we needed the struct variant State::Failed { x:y}

Reply:
I think you are also right too. I feel a bit weird for this message.
We (and rustc) should rethink about this message :)